api/src/main/java/org/openmrs/module/ModuleFactory.java

```diff
@@ -574,7 +565,7 @@ public static Module getModuleByPackage(String modulePackage) {
 	 * @see Daemon#startModule(Module)
 	 */
 	public static Module startModule(Module module) throws ModuleException {
-		return Daemon.startModule(module);
```

Comment:
@dkayiwa i have done some checking again in the code and i realised that if i don't do that change as i did, when someone tries to start a module using ModuleFactory.startModule(module), following the method calls shows that it will end up in [ModuleFactory.startModuleInternal(Module, Boolean, ApplicationContext)](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/context/Daemon.java#L64), therefore completely bypassing my changes in the other method. I think this is what i was trying to avoid when i wrote the code, please check again and see if am right.